src/Microsoft.AspNet.Mvc.Razor/Razor/PreCompileViews/RazorPreCompiler.cs

```diff
@@ -15,39 +18,44 @@ namespace Microsoft.AspNet.Mvc.Razor
 {
     public class RazorPreCompiler
     {
+        private static readonly int MaxDegreesOfParallelism = Environment.ProcessorCount;
         private readonly IServiceProvider _serviceProvider;
         private readonly IFileSystem _fileSystem;
-        private readonly IMvcRazorHost _host;
+        private readonly Func<ICacheSetContext, PrecompilationCacheEntry> _onCacheMiss;
 
         public RazorPreCompiler([NotNull] IServiceProvider designTimeServiceProvider,
+                                [NotNull] IMemoryCache precompilationCache,
```

Comment:
It is configured differently than the one in DI. Plus using specific ones avoid namespacing keys.